### DISCUSSION TITLE
serialEvent()

### DISCUSSION THREAD

**Discussion by phlebos** [NicoHood/HoodLoader2#92] [Q&A]
Hello NicoHood

First thanks.

I'm not sure if this is beyond the scope.  But I am trying to use your bootloader with  a standalone 32u2 and the Arduino IDE.
Your code has already helped a lot, basic arduino sketches working ,blink etc.  It verified my homemade lashup is working.

But I am trying to use the Arduino serialEvent() with the usbserial device presented as ttyACM0.  
It never triggers.  Should I expect it to?

Thanks

**Comment by NicoHood** [comment]
Did you try a 32u4? I am not sure, but this should be an arduino feature.

**Reply by phlebos** [reply]
Hi 
Hmm.  I think that set me on the right track.  I found the following post.
https://forum.arduino.cc/t/serialevent-on-dues-native-usb-port-never-gets-fired/200495/3

Although its an old post it suggests serialevent isn't implemented for USB type ports.
and offers a workaround I haven't tried it yet but it sounds exactly like my problem.

Cheers.
